### PR TITLE
ENG-13390 fix catalog diff engine to evaluate the security user change with catalog update

### PR DESCRIPTION
**Pull request by zeemanhe**
HTTP sessions will be cleaned up only if security users are updated in catalog update.

### CODE REVIEW

File: src/catgen/in/javasrc/CatalogDiffEngine.java [hunk 1]
```diff
@@ -241,6 +241,11 @@ public boolean requiresNewExportGeneration() {
         return true;
     }
 
+    public boolean hasSecurityUserChanges() {
+        CatalogChangeGroup ccg = m_changes.get(DiffClass.USER);
```
Comment (akhanzode) [inline src/catgen/in/javasrc/CatalogDiffEngine.java]: What about security enabled="true|false" boolean that should be checked too.

File: src/catgen/spec.txt [hunk 2]
```diff
@@ -91,6 +91,7 @@ end
 
 begin User javaonly
   GroupRef* groups
+  string password "unencrypted password"
```
Comment (akhanzode) [inline src/catgen/spec.txt]: Why do you need to add cleartext password in catalog this is bad as customers who will send us catalog will end up sending their passwords too.

**Comment by akhanzode** [comment]
Add tests in TestCatalogDiff see verifyDiff()

File: src/catgen/in/javasrc/CatalogDiffEngine.java (hunk 1 above)
Comment (zeemanhe) [reply src/catgen/in/javasrc/CatalogDiffEngine.java]: a check has been added

File: src/catgen/spec.txt (hunk 2 above)
Comment (zeemanhe) [reply src/catgen/spec.txt]: passwords are encoded with random seeds. added a encrypted  version with fixed salt for easy comparison.

File: src/catgen/spec.txt [hunk 3]
```diff
@@ -91,6 +91,7 @@ end
 
 begin User javaonly
   GroupRef* groups
+  string password "encrypted version"
```
Comment (xinjiacs) [inline src/catgen/spec.txt]: what's the "encrypted version"?

“encrypted password version for catalog comparison usage"?

File: src/frontend/org/voltdb/utils/CatalogUtil.java [hunk 4]
```diff
@@ -2172,7 +2172,8 @@ private static void setUsersInfo(Catalog catalog, UsersType users) throws Runtim
                             saltGen);
             catUser.setShadowpassword(hashedPW);
             catUser.setSha256shadowpassword(hashedPW256);
-
+            //use fixed seed for comparison
+            catUser.setPassword( BCrypt.hashpw(sha1hex, "$2a$10$pWO/a/OQkFyQWQDpchZdEe"));
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/utils/CatalogUtil.java]: Use sha256 password as we are going to remove sha1 soon.

**Pull request merged**